platform/kindle/koreader.sh

```diff
@@ -1,4 +1,15 @@
 #!/bin/sh
+
+# NOTE: Stupid workaround to make sure the script we end up running is a *copy*,
+# living in a magical land that doesn't suffer from gross filesystem deficiencies.
+# Otherwise, the vfat+fuse mess means an OTA update will break the script on exit,
+# and potentially leave the user in a broken state, with the WM still paused...
+if [ "$(dirname "${0}")" != "/var/tmp" ]; then
```

Comment:
Does it need to be preserved on reboot?

Reply:
It's a tmpfs, so it'll go poof on reboot.

I'm fine with leaving however kb this takes when we exit, it's 32MB by default, we've got more than enough space in there ;).

Reply:
Although, since it's no longer on a crappy FS, it also wouldn't hurt to unlink on exit, now that I think about it ;).

Reply:
It will? Hmm… odd.

https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch05s15.html
>5.15. /var/tmp : Temporary files preserved between system reboots
>5.15.1. Purpose
>
>The /var/tmp directory is made available for programs that require temporary files or directories that are preserved between system reboots. Therefore, data stored in /var/tmp is more persistent than data in /tmp.
>
>Files and directories located in /var/tmp must not be deleted when the system is booted. Although data stored in /var/tmp is typically deleted in a site-specific manner, it is recommended that deletions occur at a less frequent interval than /tmp.

Reply:
Anyway, I guess I mean that /var/tmp seems needlessly longer than /tmp if there's no special reason to prefer /var/tmp

Reply:
I don't have the Kindle plugged in for details, but /var is a tmpfs on Kindle, and /tmp is either not one at all, or another, much smaller one, and that one might be noexec in some setups.

So, yeah, /var/tmp should be a safer bet than /tmp, on Kindle at least ;).

(FWIW, both /tmp and /var/tmp are tmpfs on my desktop ^^).

Reply:
The TL;DR being: That was a concerted decision after checking the state of /tmp vs. /var/tmp on various Kindle generations ;).